classes/user/report/Report.inc.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * @defgroup lib_pkp_classes_user
+ */
+
+/**
+ * @file lib/pkp/classes/user/report/Report.inc.php
+ *
+ * Copyright (c) 2003-2020 Simon Fraser University
+ * Copyright (c) 2003-2020 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file LICENSE.
+ *
+ * @class Report
+ * @ingroup lib_pkp_classes_user
+ *
+ * @brief Responsible to retrieve and provide users data to the Mapping objects.
+ */
+
+namespace PKP\User\Report;
+
+import('lib.pkp.classes.user.UserDAO');
+
+class Report implements \IteratorAggregate {
+	/** @var Mapping[] An array of Mapping objects responsible to feed the report's header and body */
+	private $_mappings = [];
+
+	/** @var QueryBuilder QueryBuilder instance */
+	private $_queryBuilder;
+
+	/**
+	 * Constructor
+	 * @param bool $addDefaultMappings Whether default mappings should be automatically added (defaults to true)
+	 */
+	public function __construct(?bool $addDefaultMappings = true)
+	{
+		$this->_queryBuilder = new QueryBuilder();
+
+		if ($addDefaultMappings) {
+			new Mappings\Standard($this);
+			new Mappings\UserGroups($this);
+			new Mappings\Notifications($this);
```

Comment:
Just wrote this to have a working report by default.
I've created these `Mappings\*` in the end, just to organize a bit the fields... So they are kind of disconnected from the implementation, if needed I can add an `addUserGroups()/addMappingGroup($mapping)`...